Sources/Common/Communication/EventBus.swift

```diff
@@ -53,12 +51,6 @@ class EventBusObserversHolder {
         }
         observers.removeAll()
     }
-
-    /// Deinitializer for EventBusObserversHolder.
-    /// Ensures that all observers are removed from NotificationCenter upon deinitialization.
-    deinit {
-        self.removeAllObservers()
-    }
```

Comment:
it ensures we automatically removes all observers from notification center upon deinitialization

Reply:
I removed it because [it doesn't get called at the moment](https://linear.app/customerio/issue/MBL-208/possible-memory-leak-in-datapipelinesimplementation). Because this commit is manually calling `removeAllObservers()`, I didn't want this `deinit` in the source code to make us think that it's working when it's not.